### PR TITLE
Fix issue with generate property being called on a nested object initializer

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/13166

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs [hunk 1]
```diff
@@ -1365,6 +1365,36 @@ void M()
...
+@"
+public class Inner { public int InnerValue { get; internal set; } }
+public class Outer
+{
+    public Inner Inner { get; set; } = new Inner();
+    public static Outer X() => new Outer { Inner = { InnerValue = 5 } };
+}");
+        }
+
+        /*
```
Comment (dpoeschl) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs]: What's going on with this comment?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs]: which comment?

Comment (dpoeschl) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs]: lol

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs [hunk 2]
```diff
@@ -1365,6 +1365,27 @@ void M()
 @"class Foo { public int Gibberish { get ; internal set ; } } class Bar { void foo ( ) { var c = new Foo { Gibberish = 24 } ; } } ");
         }
 
+        [WorkItem(13166, "https://github.com/dotnet/roslyn/issues/13166")]
```
Comment (dpoeschl) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs]: VB test?

File: src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb [hunk 3]
```diff
@@ -564,8 +564,21 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
+        End Function
+
+        Public Function IsObjectInitializerNamedAssignmentIdentifier(
+                node As SyntaxNode,
+                ByRef initializedInstance As SyntaxNode) As Boolean Implements ISyntaxFactsService.IsObjectInitializerNamedAssignmentIdentifier
+
             Dim identifier = TryCast(node, IdentifierNameSyntax)
-            Return If(identifier?.IsChildNode(Of NamedFieldInitializerSyntax)(Function(n) n.Name), False)
+            If identifier?.IsChildNode(Of NamedFieldInitializerSyntax)(Function(n) n.Name) Then
+                initializedInstance = identifier.Parent.Parent.Parent
```
Comment (dpoeschl) [inline src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb]: Consider: a comment with a code example so you can easier understand .Parent.Parent.Parent

**Comment by dpoeschl** [comment]
:+1:

File: src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb (hunk 3 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb]: k.

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateVariable/GenerateVariableTests.cs]: VB does not have nested object initializers.  And it has tests for normal object initializers.

File: src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb [hunk 4]
```diff
@@ -564,8 +564,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         Public Function IsObjectInitializerNamedAssignmentIdentifier(node As SyntaxNode) As Boolean Implements ISyntaxFactsService.IsObjectInitializerNamedAssignmentIdentifier
+            Dim unused As SyntaxNode = Nothing
+            Return IsObjectInitializerNamedAssignmentIdentifier(node, unused)
+        End Function
+
+        Public Function IsObjectInitializerNamedAssignmentIdentifier(
```
Comment (jmarolf) [inline src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb]: Even if there is no current consumer of the VB side of this API, can we still write a test for it?

Comment (CyrusNajmabadi) [inline src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb]: There are VB consumers of both these APIs.  GenerateVariable ends up using both.  It uses the first to determine if it should offer to GenerateLocal or not, and it uses hte latter to decide what type to actually generate into.  :)

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**